adddish_sql_op.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+include "connectDB_localhost.php";  
+
+$item	= $_POST["itemname"];
+$inventory	= $_POST["inventory"];
+$sp = $_POST["selling_price"]; //selling price.
+$bp = $_POST["buying_price"]; //buying price.
+$img_p = $_POST["image_path"]; //image path.
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Add input validation for POST data</b></div>
<div id="fix">

Input validation is missing for POST parameters. Consider validating the data types and ranges before processing.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -5,9 +5,17 @@
 -$item	= $_POST["itemname"];
 -$inventory	= $_POST["inventory"];
 -$sp = $_POST["selling_price"];
 -$bp = $_POST["buying_price"];
 -$img_p = $_POST["image_path"];
 +if(!isset($_POST["itemname"]) || empty($_POST["itemname"])) {
 +    die("Item name is required");
 +}
 +if(!is_numeric($_POST["inventory"]) || $_POST["inventory"] < 0) {
 +    die("Invalid inventory value");
 +}
 +if(!is_numeric($_POST["selling_price"]) || $_POST["selling_price"] <= 0) {
 +    die("Invalid selling price");
 +}
 +$item = htmlspecialchars($_POST["itemname"]);
 +$inventory = (int)$_POST["inventory"];
 +$sp = (float)$_POST["selling_price"];
 +$bp = (float)$_POST["buying_price"];
 +$img_p = htmlspecialchars($_POST["image_path"]);
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Lux-Jason/rms_new/pull/8#issuecomment-2544501398>#de856c</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged